### DISCUSSION TITLE
Tailwindcss can be working in component of Vue 3 with the style via a variable parameter?

### DISCUSSION THREAD

**Discussion by carveybunt** [tailwindlabs/tailwindcss#4813] [Ideas]
How tailwindcss uses mode `jit` to build the CSS file if the class name uses a variable parameter in the component of Vue3 via the prop?
My `postcss.config.js`:
```js
module.exports = {
  plugins: {
    tailwindcss: {},
    autoprefixer: {},
  }
}
```
My `tailwind.config.js`:
```js
module.exports = {
  mode:'jit',
  purge: [
    ...
    'components/**/*.{vue,js,ts,jsx,tsx}',
    ...
  ],
 ...
}
```
`tailwind.css`:
```js
@import "tailwindcss/base";
@import "tailwindcss/components";
@import "tailwindcss/utilities";
```
For example, my component of Vue 3:
```js
<template>
...
<p v-if="$slots.default || !!description" :class="[ ..., descriptionSize ? 'text-'+descriptionSize : 'text-xs']">
  <slot>
    {{ description }}
  </slot>
</p>
...
</template>

<script lang='ts'>
...
props: {
  ...
  descriptionSize:{
    type: String,
    default: '',
  },
  ...
}
...
</script>
```
When I use my component in the Vue file, also prop the descriptionSize as `lg/sm/xs` too. The `text-lg/text-sm/text-xl` not working in the `tw.css` when I running `npx tailwindcss-cli@latest build ./tailwind.css -o ./assets/tw.css`. Could someone help me, please?

**Comment by doommm** [comment]
because your current mode is `jit`, and in [tailwind's documentation](https://tailwindcss.com/docs/just-in-time-mode#arbitrary-value-support), it mentions that **Don't use string concatenation to create class names, Do dynamically select a complete class name**

**Reply by carveybunt** [reply]
Thanks.

**Reply by HichamDouch** [reply]
Thanks  @doommm

**Comment by VicGUTT** [comment]
Hey ! As mentioned above, you should **not** do string interpolation like that and should instead do something like:

```html
<p v-if="$slots.default || !!description" :class="[ ..., descriptionSizeClass]">
```

```js
{
    computed: {
        descriptionSizeClass() {
            switch (this.descriptionSize) {
                case 'xs':
                    return 'text-xs';
                case 'md':
                    return 'text-md';
            case ...:
                    ...
                default:
                    return 'text-xs';
            }
        },
    },
}
```

Or even the following should work:

```html
<p v-if="$slots.default || !!description" :class="[ ..., descriptionSize ? 'text-'+descriptionSize : 'text-xs']">
<!-- text-xs text-md ... -->
```

The reason being that JIT mode uses purgeCSS under the hood to find and parse your vue component in search of any string Tailwind knows to be a utility class.
Therefore the string `'text-'+descriptionSize` is not a Tailwind utility, so it is discarded unless it finds `text-xs, text-md, ...` anywhere else.

**Reply by carveybunt** [reply]
Thanks a lot, but this isn't my wanna, because my script in my component not just simple like my issue. For example for a section of my component:
```js
const TYPE_CLASSES_MAP = {
  'success': 'green',
  'warning': 'yellow',
  'error': 'red',
  'info': 'gray',
}
const setBgColor = ()=>{
  const color = props.color
  return 'bg-'+color+'-100 hover:bg-'+ color + '-200 dark:bg-'+color+'-400 dark:hover:bg-'+ color + '-500 '
}
const setTypeColor = (str,drak)=>{
  const type = TYPE_CLASSES_MAP[props.type]
  if(drak){
    return str+type+'-100 hover:'+str+type+'-200 dark:'+str+type+'-400 dark:hover:'+str+type+'-500'
  }
  return str+type+'-400 hover:'+str+type+'-500 dark:'+str+type+'-100 dark:hover:'+str+type+'-200'
}
const typeClassFn =computed(()=>{
  const style = props.styles
  const color = props.color
  if(!style && !color){
    return setTypeColor('bg-',1)
  }
  if (!style && color){
    return setBgColor()
  }
  return style
})
``` 
This allows the component to be rendered dynamically via the type and color parameters why I do this. I already change my mind that I decided to use the class name of tailwindcss now.

**Comment by josueayala27** [comment]
I recommend you have an object with properties (They can be `key`, `value`) where you define the classes for each key 👽 for example:
`data: { 
     styles = { blue: "bg-blue-500 text-white...", otherProp: "bg-orange..." }
     ...
}`

So you can dynamically call through some `computed` property:

```
getStyles() {
     return this.styles[key];
     // If u key is blue, return the first element of you object 🙃
}
```

This is a vague example, but basically that's how it would work ✅
